packages/web/src/flash/FlashReducer.ts

```diff
@@ -0,0 +1,63 @@
+export interface FlashMessage {
+  id: number
+  text: string
+  style?: React.HTMLAttributes<HTMLDivElement>['style']
+  classes?: string
+  persist?: boolean
+  viewed?: boolean
+}
+
+type FlashAction =
+  | {
+      type: 'ADD_MESSAGE'
+      message: Omit<FlashMessage, 'id'>
+    }
+  | {
+      type: 'DISMISS_MESSAGE'
+      messageId: number
+    }
+  | {
+      type: 'CYCLE_MESSAGE'
+      messageId: number
+    }
+
+// the reducer
+export default (messages: FlashMessage[] = [], action: FlashAction) => {
+  switch (action.type) {
+    case 'ADD_MESSAGE': {
+      const {
+        text,
+        style = {},
+        classes = '',
+        persist = false,
+        viewed = false,
+      } = action.message
+
+      const newMessage = {
+        id: messages.length,
+        text,
+        style,
+        classes,
+        persist,
+        viewed,
+      }
```

Comment:
I don't understand why all this is needed. Why can't you just do like this?
```suggestion
      const newMessage = {...action.message, id: messages.length}
```

Reply:
@Tobbe because of the default values